app/src/main/java/com/canopas/yourspace/ui/flow/home/space/create/SpaceInviteCodeScreen.kt

```
@@ -68,7 +76,15 @@ fun SpaceInvite() {
             },
             actions = {
                 if (state.isUserAdmin) {
-                    IconButton(onClick = { viewModel.regenerateInviteCode() }) {
+                    IconButton(
+                        onClick = {
+                            if (state.connectivityStatus == ConnectivityObserver.Status.Available) {
+                                viewModel.regenerateInviteCode()
+                            } else {
+                                viewModel.setErrorState(Exception())
```

detekt (1.23.7)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)